waiter/src/waiter/metrics_sync.clj

```diff
@@ -343,11 +348,34 @@
         metrics-agent (agent initial-state)]
     metrics-agent))
 
+(defn- merge-router-metrics
```

Comment:
How about factoring out something like `merge-by` (similar to`merge-with`) that also takes the key as an argument?

<pre>
(merge-by
  (fn [key v1 v2] ...)
  m1 m2)
</pre>